plugins/modules/ndo_tenant_span_session.py

```diff
@@ -0,0 +1,587 @@
+    argument_spec.update(
+        template=dict(type="str"),
+        template_id=dict(type="str"),
+        name=dict(type="str", aliases=["span_session"]),
+        uuid=dict(type="str", aliases=["span_session_uuid"]),
+        description=dict(type="str"),
+        sources=dict(
+            type="list",
+            elements="dict",
+            mutually_exclusive=[
+                ("epg", "epg_uuid"),
+            ],
+            required_one_of=[
+                ["epg", "epg_uuid"],
+            ],
+            options=dict(
+                name=dict(type="str", required=True),
+                direction=dict(type="str", required=True, choices=["incoming", "outgoing", "both"]),
+                epg_uuid=dict(type="str"),
+                epg=epg_object_reference_spec(),
+            ),
+        ),
+        admin_state=dict(type="str", choices=list(ENABLED_OR_DISABLED_TO_BOOL_STRING_MAP)),
+        mtu=dict(type="int"),
+        destination_epg=dict(
+            type="dict",
+            mutually_exclusive=[
+                ("epg", "epg_uuid"),
+            ],
+            options=dict(
+                epg_uuid=dict(type="str"),
+                epg=epg_object_reference_spec(),
+                destination_ip=dict(type="str"),
+                source_ip_prefix=dict(type="str"),
+                span_version=dict(type="str", choices=["v1", "v2"]),
+                enforce_span_version=dict(type="bool"),
+                flow_id=dict(type="int"),
+                ttl=dict(type="int"),
+                dscp=dict(type="str", choices=list(TARGET_DSCP_MAP)),
+            ),
+        ),
+        state=dict(type="str", default="query", choices=["absent", "query", "present"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        mutually_exclusive=[
+            ("template", "template_id"),
+        ],
+        required_if=[
+            ["state", "absent", ["name", "uuid"], True],
+            ["state", "present", ["name", "uuid"], True],
+        ],
+        required_one_of=[["template", "template_id"]],
+    )
+
+    mso = MSOModule(module)
+    mso_schema = None
+
+    template_name = module.params.get("template")
+    template_id = module.params.get("template_id")
+    name = module.params.get("name")
+    uuid = module.params.get("uuid")
+    description = module.params.get("description")
+    sources = module.params.get("sources")
+    admin_state = module.params.get("admin_state")
+    mtu = module.params.get("mtu")
+    destination_epg = module.params.get("destination_epg")
+    state = module.params.get("state")
+
+    ops = []
+    match = None
+
+    mso_template = MSOTemplate(mso, "monitoring_tenant", template_name, template_id)
+    mso_template.validate_template("monitoring")
+    object_description = "SPAN Session"
+    path = "/monitoringTemplate/template/spanSessions"
+    span_session_path = None
+    reference_collection = {
```

Comment:
why specify this here and not as part of the function set_template_and_references which seems to be the only location where it is used? is this something that should be defined in constants? perhaps combined with other references?

Reply:
It made sense to me to keep all module specific constants defined a single time instead of in each `set_template_and_references` call. However, I have moved it into the function to simplify the code.
For now I have kept it consistent with our existing PRs. I think we can make these `reference_collection` definitions shared constants when we start writing the same config multiple times.

Reply:
only then we do not know it is already used somewhere, but fine by me